graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/InsertFetcher.java

```diff
@@ -70,75 +73,105 @@ protected Object get(DataFetchingEnvironment environment, StargateGraphqlContext
-            .collect(Collectors.toList());
-
-    Optional<Long> timestamp =
-        TimestampParser.parse(model.getCqlTimestampArgumentName(), environment);
-
-    AbstractBound<?> query =
-        context
-            .getDataStore()
-            .queryBuilder()
-            .insertInto(entityModel.getKeyspaceName(), entityModel.getCqlName())
-            .value(modifiers)
-            .ifNotExists(isLwt)
-            .ttl(model.getTtl().orElse(null))
-            .timestamp(timestamp.orElse(null))
-            .build()
-            .bind();
-
-    context
-        .getAuthorizationService()
-        .authorizeDataWrite(
-            context.getSubject(),
-            entityModel.getKeyspaceName(),
-            entityModel.getCqlName(),
-            TypedKeyValue.forDML((BoundDMLQuery) query),
-            Scope.MODIFY,
-            SourceAPI.GRAPHQL);
-
-    ResultSet resultSet = executeUnchecked(query, buildParameters(environment), context);
-
-    if (responseContainsEntity) {
-      Map<String, Object> entityData;
-      if (entityPrefixInReponse == null) {
-        entityData = response;
+    List<Map<String, Object>> inputs;
+    List<Map<String, Object>> responses = new ArrayList<>();
+    // needed if the return type is a List of Booleans
+    List<Boolean> appliedList = new ArrayList<>();
+    if (model.isList()) {
+      inputs = environment.getArgument(model.getEntityArgumentName());
+    } else {
+      inputs = Collections.singletonList(environment.getArgument(model.getEntityArgumentName()));
+    }
+
+    boolean applied = false;
+    for (Map<String, Object> input : inputs) {
+      Map<String, Object> response = new LinkedHashMap<>();
+      Map<String, Object> cqlValues = buildCqlValues(entityModel, keyspace, input);
+      Collection<ValueModifier> modifiers =
+          cqlValues.entrySet().stream()
+              .map(e -> ValueModifier.set(e.getKey(), e.getValue()))
+              .collect(Collectors.toList());
+
+      Optional<Long> timestamp =
+          TimestampParser.parse(model.getCqlTimestampArgumentName(), environment);
+
+      AbstractBound<?> query =
+          context
+              .getDataStore()
+              .queryBuilder()
+              .insertInto(entityModel.getKeyspaceName(), entityModel.getCqlName())
+              .value(modifiers)
+              .ifNotExists(isLwt)
+              .ttl(model.getTtl().orElse(null))
+              .timestamp(timestamp.orElse(null))
+              .build()
+              .bind();
+
+      context
+          .getAuthorizationService()
+          .authorizeDataWrite(
+              context.getSubject(),
+              entityModel.getKeyspaceName(),
+              entityModel.getCqlName(),
+              TypedKeyValue.forDML((BoundDMLQuery) query),
+              Scope.MODIFY,
+              SourceAPI.GRAPHQL);
+
+      ResultSet resultSet = executeUnchecked(query, buildParameters(environment), context);
+
+      if (responseContainsEntity) {
```

Comment:
done, please take a look. 
I am not sure what do you mean by ` We can pass in entityPrefixInReponse, with null meaning that it's not present.` I think that the `entityPrefixInReponse` needs to be passed there, in case the user provides its own prefix.

Reply:
Nice, this definitely improves readability. But it can be simplified even further:

If we go into `buildEntityResponse()`, `entityPrefixInResponse` is always null (the response is not a payload => the entity cannot be a subfield).
So you don't need to pass it in and you can simplify the statements that tested it. Also, you know that the response will always be a simple map (no nesting), so I don't think we need the intermediary `copyInputDataToResponse` method:
```java
  private Map<String, Object> buildEntityResponse(
      boolean isLwt,
      ResultSet resultSet,
      Map<String, Object> input,
      Map<String, Object> cqlValues) {

    Map<String, Object> entityResponse = new LinkedHashMap<>();
    copyInputDataToResponse(input, cqlValues, entityResponse, model.getEntity());

    if (isLwt) {
      Row row = resultSet.one();
      if (!row.getBoolean("[applied]")) {
        copyRowToEntity(row, entityResponse, model.getEntity());
      }
    }
    return entityResponse;
  }
```

Conversely, in `buildPayloadResponse`, we know that if the entity is present, it is always nested. However it may also be absent (if the payload only contains the applied field). This is when we can have `entityPrefixInReponse == null`:
```java
...
    Map<String, Object> response = new LinkedHashMap<>();
    Map<String, Object> entityInResponse = null;
    if (entityPrefixInResponse != null) {
      entityInResponse = new LinkedHashMap<>();
      copyInputDataToResponse(input, cqlValues, response, model.getEntity());
      response.put(entityPrefixInResponse, entityInResponse);
    }

    boolean applied;
    if (isLwt) {
      Row row = resultSet.one();
      applied = row.getBoolean("[applied]");
      if (!applied && entityPrefixInResponse != null) {
        copyRowToEntity(row, entityInResponse, model.getEntity());
      }
    } else {
      applied = true;
    }
...
```

PS: you can also modify the original `copyRowToEntity` to reference `model.getEntity()` directly instead of passing it as a parameter, that was an oversight on my part.

Reply:
Nice! I didn't know that the `entityPrefixInResponse` is always null for the `buildEntityResponse`. I simplifies things a lot indeed.
Regarding:
> you can also modify the original copyRowToEntity to reference model.getEntity() directly instead of passing it as a parameter, that was an oversight on my part.

I think you meant: `copyInputDataToResponse`, not `copyRowToEntity`. The latter is in the `DeployedFetcher` and does not have access to the `getEntity` method. Done.

Reply:
> I think you meant: `copyInputDataToResponse`, not `copyRowToEntity`

Yes

Reply:
does it look good now? Can I merge?